### DISCUSSION TITLE
Fine grained task management

### DISCUSSION THREAD

**Discussion by rezamahdi** [tokio-rs/axum#2199] [Q&A]
Hi.
I wonder that is there any utilities to manage and control the async tasks in axum?

For example set a maximum number of connections, terminating a long-running (hanged up, may be...) task and so on...

Having such a control helps creating production-efficient and more reliable apps

**Comment by davidpdrsn** [comment]
Not in axum itself. All connection stuff is handled by hyper, not axum. So you need to use [hyper's low level connection API](https://docs.rs/hyper/latest/hyper/server/conn/index.html) to customize that.
